src/WooCommerce.php

```diff
@@ -1217,6 +1217,14 @@ public static function woocommerce_save_account_details($user_id) {
       Server::addDebugMessage();
     }
     else {
+      // If a new email address has been set, replace the success message to
+      // clarify that the user needs to confirm the change via email.
+      if (isset($userinfo['action']) && $userinfo['action'] === 'changeEmail') {
+        $all_notices = WC()->session->get('wc_notices', []);
+        $index = array_search(__('Account details changed successfully.', 'woocommerce'), $all_notices['success']);
+        $all_notices['success'][$index] = 'Bitte bestätigen Sie Ihre neue E-Mail-Adresse, indem Sie den Link anklicken, den wir Ihnen soeben an Ihre neue E-Mail-Adresse gesendet haben.';
```

Comment:
> Bitte bestätigen Sie Ihre neue E-Mail-Adresse, indem Sie den Link anklicken, den wir Ihnen soeben an Ihre neue E-Mail-Adresse gesendet haben.

The proposed message needs review / feedback.

Reply:
WordPress uses:
> Ihre E-Mail-Adresse wurde noch nicht aktualisiert. Bitte prüfen Sie Ihren Posteingang für %s auf eine Bestätigungs-E-Mail.
```po
#. translators: %s: New email address
#: wp-includes/user.php:2731
msgid "Your email address has not been updated yet. Please check your inbox at %s for a confirmation email."
msgstr "Ihre E-Mail-Adresse wurde noch nicht aktualisiert. Bitte prüfen Sie Ihren Posteingang für %s auf eine Bestätigungs-E-Mail."
```

WooCommerce does not require a new email address to be confirmed (which is why we did not have to disable or override a built-in functionality).

To my surprise, [Drupal does not have any strings for this](https://localize.drupal.org/translate/languages/de/translate?project=drupal&status=0&release=488912&search=email%20address&author&context=all&limit=50&sid=0) either.

Reply:
Moving forward with a mixture of the current string and the one from WordPress Core:
> Ihre E-Mail-Adresse wurde noch nicht aktualisiert. Bitte prüfen Sie Ihren Posteingang für %s und klicken Sie den Link zur Bestätigung Ihrer neuen E-Mail-Adresse.